File: OTI/TP1-js/money_js/src/money.js
var money=function(v,curr) {
	this.v=v;
	this.curr=curr;
}

money.prototype.getCurrency=
        function () {
		return this.curr.toUpperCase();
	};

money.prototype.getValue=
        function () {
		return this.v;
	};

money.prototype.equals=
	function (otherM) {
		return (otherM.getValue()==this.getValue())&&(otherM.getCurrency()==this.getCurrency());
	};
        
money.prototype.toString=
	function () {
		return this.getValue()+" ("+this.getCurrency()+")" ;
	};

File: OTI/TP1-JUnit/src/fr/ulille1/fil/odeva/Money.java
package fr.ulille1.fil.odeva;

import java.util.HashMap;
import java.util.Map;

public class Money {

    private int value;
    private String currency;

    Money(int value, String currency) {
        this.value = value;
        this.currency = currency;
    }

    public int getValue() {
        return this.value;
    }

    public String getCurrency() {
        return this.currency;
    }

    public String toString() {
        return this.getValue() + " (" + this.getCurrency() + ")";
    }
    
    public boolean equals(Object o){
        if(!(o instanceof Money)){
            return false;
        }
        Money m = (Money) o;
        return (m.getValue() == this.getValue())&&(m.getCurrency().equals(m.getCurrency()));
    }

}
